File: content/cpp/templates/function-params.cpp
// Function with Default Parameters

#include <iostream>
#include <string>
using namespace std;

string greet(string name, string greeting = "Hello") {
    return greeting + ", " + name + "!";
}

int main() {
    cout << greet("Alice") << endl;
    cout << greet("Bob", "Hi") << endl;
    
    return 0;
}

File: content/java/templates/05_Functions/function-params.java
// Method with Overloading (Java alternative to default params)

public class FunctionParams {
    // Overloaded methods (Java way of default parameters)
    public static String greet(String name) {
        return greet(name, "Hello");
    }
    
    public static String greet(String name, String greeting) {
        return greeting + ", " + name + "!";
    }
    
    public static void main(String[] args) {
        System.out.println(greet("Alice"));
        System.out.println(greet("Bob", "Hi"));
    }
}
